block/internal/syncing/syncer.go

```diff
@@ -284,17 +284,24 @@ func (s *Syncer) syncLoop() {
 			} else if len(events) > 0 {
 				// Reset backoff on success
 				nextDARequestAt = time.Time{}
+
+				eventDropped := false
+
 				// Process DA events
 				for _, event := range events {
 					select {
 					case s.heightInCh <- event:
 					default:
 						s.logger.Warn().Msg("height channel full, dropping DA event")
+						eventDropped = true
+						break
```

Comment:
alternatively, you could also use a `continue` with an outer label .

Reply:
Optional: As the consumers on the `heightInCh` channel are slow in this scenario, you can buy them some time with a `sleep` before continuing with the for-loop

Reply:
i like it, it remove those need of those temporary variables.